scipy/signal/tests/test_signaltools.py

```diff
@@ -567,6 +567,14 @@ def test_polyphase_extfilter(self):
         # Test external specification of downsampling filter
         self._test_data(method='polyphase', ext=True)
 
+    def test_mutable_window(self):
+        # Test that a mutable window is not modified
+        impulse = np.zeros(3)
+        window = np.random.RandomState(0).randn(2)
+        window_orig = window.copy()
+        signal.resample_poly(impulse, 5, 1, window=window)
+        np.testing.assert_allclose(window, window_orig, atol=1e-7)
```

Comment:
This test should use `assert_equal()`.   There is no reason for *any* change to the array to be acceptable here.